autoscaler/controllers/datacollection/daemonset.go

```diff
@@ -369,3 +379,29 @@ func patchDaemonSet(existing *appsv1.DaemonSet, desired *appsv1.DaemonSet, ctx c
 
 	return updated, nil
 }
+
+func isVersionSupported(cfg *rest.Config, minMajor int, minMinor int) (bool, error) {
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(cfg)
+	if err != nil {
+		return false, err
+	}
+
+	serverVersion, err := discoveryClient.ServerVersion()
+	if err != nil {
+		return false, err
+	}
```

Comment:
Since this will be called on every reconcile, we need to avoid this re-calculation and querying the API server and do it once in main.